shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java

```diff
@@ -229,4 +229,27 @@ public static String failMetricName(String metricName) {
             return metricName + "_fail";
         }
     }
+
+    /**
+     * Generate a dot separated combination of metric name and tag value[s].
+     * e.g. metric "pravega.segmentstore.container.append_count" with tag "container=6" will get:
+     * "pravega.segmentstore.container.append_count.6".
+     *
+     * @param metric the metric name.
+     * @param tags the tag(s) associated with the metric.
+     * @return the dot separated combination of metric name and tag value(s).
+     */
+    public static String nameFromTags(String metric, String... tags) {
+        if (tags == null || tags.length == 0) {
+            return metric;
+        }
+        if (tags.length % 2 == 1) {
```

Comment:
Use `Preconditions.checkArgument`

Reply:
Tags could be null or empty, it is not illegalArgument.